packages/visual-editor/src/hooks/useEditorSubscriber.spec.ts

```diff
@@ -85,19 +146,42 @@ describe('Canvas Subscriber methods', () => {
 
       // assert that the listener is called with the expected payload and performs the right actions
       expect(listener).toBeDefined();
-      listener!(new MessageEvent('message', { data: JSON.stringify(payload) }));
-      expect(updateNodesByUpdatedEntity).toHaveBeenNthCalledWith(1, updatedEntity.sys.id);
-      expect(updateEntity).toHaveBeenNthCalledWith(1, updatedEntity);
+      await act(() => listener!(new MessageEvent('message', { data: JSON.stringify(payload) })));
+
+      expectations?.(payload.payload);
     });
 
-  it.each`
-    event    | payload           | data
-    ${event} | ${messagePayload} | ${data}
-  `('should receive the expected payload for "$event" event', async ({ event, payload, data }) => {
-    await pact
-      .given(`a ${event} message is sent`)
-      .expectsToReceive(`a ${event} message`)
-      .withContent(data)
-      .verify(createPostMessageReceiver(event, payload));
-  });
+  const createPostMessageReceiverForSelectedEntities = (_event: IncomingEvent, payload) =>
+    asynchronousBodyHandler(async () => {
+      const store = new EditorModeEntityStore({ entities: [], locale: 'en-US' });
+      console.log(store);
+
+      const promise = store.fetchEntities({
+        missingEntryIds: [entityIds.ENTRY1, entityIds.ENTRY2],
+        missingAssetIds: [],
+        skipCache: true,
+      });
+      window.dispatchEvent(new MessageEvent('message', { data: JSON.stringify(payload) }));
+
+      await promise;
+
+      expect(store.entities).toEqual(expect.arrayContaining(entries));
+    });
+
+  it.each(interactions)(
+    'should receive the expected payload for $event event',
+    async ({ id, description, event, payload, payloadMatcher }) => {
+      let messageReceiverConstructor = createPostMessageReceiver;
+      if (event !== INCOMING_EVENTS.RequestedEntities) {
+        return;
+      }
```

Comment:
Looks like we're currently testing only one out of all those interactions. Is that intended?